### PR TITLE
Reduce rebuild times when invoking 'et run'

### PR DESCRIPTION
**Pull request by johnmccutchan**
Android only for now but this avoids rebuilding a bunch of test targets that aren't necessary to do a `flutter run` with a custom engine.

I need to fill in some blanks for other platforms before landing.

### CODE REVIEW

**Comment by johnmccutchan** [comment]
Build time for a single line change to Android Engine

before: 76 seconds
after: 41 seconds

1.85x speedup

File: tools/engine_tool/lib/src/commands/run_command.dart [hunk 1]
```diff
@@ -112,14 +113,22 @@ See `flutter run --help` for a listing
...
   }
 
+  Future<RunTarget?>? _runTarget;
+  Future<RunTarget?>? _getRunTarget() {
+    if (_runTarget != null) {
+      return _runTarget;
+    }
+    _runTarget = detectAndSelectRunTarget(environment, _getDeviceId());
+    return _runTarget;
+  }
```
Comment (zanderso) [inline tools/engine_tool/lib/src/commands/run_command.dart:123]: This is one of the use cases for `late final`, but I don't remember off the top of my head if you need to wrap it in a closure. It's either this:

```dart
  late final Future<RunTarget?> _runTarget = detectAndSelectRunTarget(environment, _getDeviceId());
```

Or this:
```dart
  late final Future<RunTarget?> _runTarget = (){
    return detectAndSelectRunTarget(environment, _getDeviceId());
  }();
```

File: tools/engine_tool/lib/src/gn.dart [hunk 2]
```diff
@@ -105,9 +111,10 @@ sealed class BuildTarget {
       String type,
       bool testOnly,
     ) = json.map((JsonObject json) => (
-      json.string('type'),
-      json.boolean('testonly'),
-    ));
+          json.string('type'),
+          json.boolean('testonly'),
+        ));
+    print('$type $label');
```
Comment (zanderso) [inline tools/engine_tool/lib/src/gn.dart:117]: Leftover debug print?

File: tools/engine_tool/lib/src/run_utils.dart [hunk 3]
```diff
@@ -8,6 +8,7 @@ import 'package:process_runner/process_runner.dart';
 
 import 'environment.dart';
 import 'typed_json.dart';
+import 'label.dart';
```
Comment (zanderso) [inline tools/engine_tool/lib/src/run_utils.dart:11]: Alphabetize imports.

File: tools/engine_tool/test/run_command_test.dart [hunk 4]
```diff
@@ -18,6 +18,7 @@ import 'package:process_fakes/process_fakes.dart';
 import 'package:process_runner/process_runner.dart';
 
 import 'fixtures.dart' as fixtures;
+import 'package:engine_tool/src/label.dart';
```
Comment (zanderso) [inline tools/engine_tool/test/run_command_test.dart:21]: I think this should go up with the other `engine_tool` imports.

File: tools/engine_tool/lib/src/commands/run_command.dart [hunk 5]
```diff
@@ -9,6 +9,7 @@ import 'package:process_runner/process_runner.dart';
 
 import '../build_utils.dart';
 import '../run_utils.dart';
+import '../label.dart';
```
Comment (zanderso) [inline tools/engine_tool/lib/src/commands/run_command.dart:12]: Alphabetize imports.

File: tools/engine_tool/lib/src/commands/run_command.dart (hunk 1 above)
Comment (matanlurey) [reply tools/engine_tool/lib/src/commands/run_command.dart:123]: If it's a single expression I believe the former should work.

**Review by matanlurey — APPROVED** [review]
Address Zach's comments but otherwise LGTM.

File: tools/engine_tool/lib/src/commands/run_command.dart (hunk 1 above)
Comment (johnmccutchan) [reply tools/engine_tool/lib/src/commands/run_command.dart:123]: Done, much nicer now!

**Pull request merged**